tests/test_utils.py

```diff
@@ -231,6 +233,15 @@ def test_instantiate_adam() -> None:
     assert res == Adam(params=adam_params)
 
 
+def test_instantiate_with_missing_module() -> None:
+
+    with pytest.raises(
+        HydraException, match=re.escape(" No module named 'some_missing_module'")
```

Comment:
1. test against the full exception message.
2. What does the exception look like in the console on this error?

Reply:
2. 
```
(hydra38) (mtrl) ➜  hydra_debug HYDRA_FULL_ERROR=1 python my_app.py
Traceback (most recent call last):
  File "/private/home/sodhani/projects/hydra/hydra/_internal/utils.py", line 522, in _locate
    import_module(mod)
  File "/private/home/sodhani/.conda/envs/hydra38/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 783, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/private/home/sodhani/hydra_debug/code/example_code.py", line 1, in <module>
    import pytorch_lightning.metrics.functional as plm
ModuleNotFoundError: No module named 'pytorch_lightning'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/private/home/sodhani/projects/hydra/hydra/utils.py", line 33, in call
    type_or_callable = _locate(cls)
  File "/private/home/sodhani/projects/hydra/hydra/_internal/utils.py", line 524, in _locate
    raise ImportError(
ImportError: Encountered error: `No module named 'pytorch_lightning'` when loading module 'code.example_code.sample_fucntion'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/private/home/sodhani/projects/hydra/hydra/_internal/utils.py", line 198, in run_and_report
    return func()
  File "/private/home/sodhani/projects/hydra/hydra/_internal/utils.py", line 333, in <lambda>
    lambda: hydra.run(
  File "/private/home/sodhani/projects/hydra/hydra/_internal/hydra.py", line 105, in run
    return run_job(
  File "/private/home/sodhani/projects/hydra/hydra/core/utils.py", line 123, in run_job
    ret.return_value = task_function(task_cfg)
  File "my_app.py", line 7, in my_app
    python_class = hydra.utils.instantiate(cfg.code)
  File "/private/home/sodhani/projects/hydra/hydra/utils.py", line 40, in call
    raise HydraException(f"Error calling '{cls}' : {e}") from e
hydra.errors.HydraException: Error calling 'code.example_code.sample_fucntion' : Encountered error: `No module named 'pytorch_lightning'` when loading module 'code.example_code.sample_fucntion'
```

Reply:
2. 
```
Traceback (most recent call last):
  File "/private/home/sodhani/projects/hydra/hydra/utils.py", line 35, in call
    return _instantiate_class(type_or_callable, config, *args, **kwargs)
  File "/private/home/sodhani/projects/hydra/hydra/_internal/utils.py", line 478, in _instantiate_class
    return clazz(*args, **final_kwargs)
  File "/private/home/sodhani/projects/hydra/tests/__init__.py", line 93, in __init__
    import some_missing_module  # type: ignore # noqa: F401
ModuleNotFoundError: No module named 'some_missing_module'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "tests/test_utils.py", line 252, in <module>
    utils.instantiate(ObjectConf(target="tests.ClassWithMissingModule"))
  File "/private/home/sodhani/projects/hydra/hydra/utils.py", line 40, in call
    raise HydraException(f"Error calling '{cls}' : {e}") from e
hydra.errors.HydraException: Error calling 'tests.ClassWithMissingModule' : No module named 'some_missing_module'
```

Reply:
1. By full expecption message, you are referring to `hydra.errors.HydraException: Error calling 'tests.ClassWithMissingModule' : No module named 'some_missing_module'` correct ?

Reply:
1. correct.
2. Is the above exception what the user sees normally or when when HYDRA_FULL_ERROR is used?

Reply:
For this example, it doesnt make any difference. Adding the output for OP's code:
```
(hydra38) (mtrl) ➜  hydra_debug HYDRA_FULL_ERROR=0 python my_app.py
Traceback (most recent call last):
  File "/private/home/sodhani/projects/hydra/hydra/_internal/utils.py", line 522, in _locate
    import_module(mod)
  File "/private/home/sodhani/.conda/envs/hydra38/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 783, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/private/home/sodhani/hydra_debug/code/example_code.py", line 1, in <module>
    import pytorch_lightning.metrics.functional as plm
ModuleNotFoundError: No module named 'pytorch_lightning'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/private/home/sodhani/projects/hydra/hydra/utils.py", line 33, in call
    type_or_callable = _locate(cls)
  File "/private/home/sodhani/projects/hydra/hydra/_internal/utils.py", line 524, in _locate
    raise ImportError(
ImportError: Encountered error: `No module named 'pytorch_lightning'` when loading module 'code.example_code.sample_fucntion'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "my_app.py", line 7, in my_app
    python_class = hydra.utils.instantiate(cfg.code)
  File "/private/home/sodhani/projects/hydra/hydra/utils.py", line 40, in call
    raise HydraException(f"Error calling '{cls}' : {e}") from e
hydra.errors.HydraException: Error calling 'code.example_code.sample_fucntion' : Encountered error: `No module named 'pytorch_lightning'` when loading module 'code.example_code.sample_fucntion'

Set the environment variable HYDRA_FULL_ERROR=1 for a complete stack trace.
```

And

```
(hydra38) (mtrl) ➜  hydra_debug HYDRA_FULL_ERROR=1 python my_app.py
Traceback (most recent call last):
  File "/private/home/sodhani/projects/hydra/hydra/_internal/utils.py", line 522, in _locate
    import_module(mod)
  File "/private/home/sodhani/.conda/envs/hydra38/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 783, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/private/home/sodhani/hydra_debug/code/example_code.py", line 1, in <module>
    import pytorch_lightning.metrics.functional as plm
ModuleNotFoundError: No module named 'pytorch_lightning'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/private/home/sodhani/projects/hydra/hydra/utils.py", line 33, in call
    type_or_callable = _locate(cls)
  File "/private/home/sodhani/projects/hydra/hydra/_internal/utils.py", line 524, in _locate
    raise ImportError(
ImportError: Encountered error: `No module named 'pytorch_lightning'` when loading module 'code.example_code.sample_fucntion'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/private/home/sodhani/projects/hydra/hydra/_internal/utils.py", line 198, in run_and_report
    return func()
  File "/private/home/sodhani/projects/hydra/hydra/_internal/utils.py", line 333, in <lambda>
    lambda: hydra.run(
  File "/private/home/sodhani/projects/hydra/hydra/_internal/hydra.py", line 105, in run
    return run_job(
  File "/private/home/sodhani/projects/hydra/hydra/core/utils.py", line 123, in run_job
    ret.return_value = task_function(task_cfg)
  File "my_app.py", line 7, in my_app
    python_class = hydra.utils.instantiate(cfg.code)
  File "/private/home/sodhani/projects/hydra/hydra/utils.py", line 40, in call
    raise HydraException(f"Error calling '{cls}' : {e}") from e
hydra.errors.HydraException: Error calling 'code.example_code.sample_fucntion' : Encountered error: `No module named 'pytorch_lightning'` when loading module 'code.example_code.sample_fucntion'
```

Reply:
Hmm, even in this case, it doesnt make a difference so I think this is the standard output.